nipype/interfaces/fsl/ICA_AROMA.py

```diff
@@ -0,0 +1,115 @@
+    TraitedSpec,
+    CommandLineInputSpec,
+    CommandLine,
+    File,
+    Directory,
+    traits,
+    OutputMultiPath
+)
+import os
+
+class ICA_AROMAInputSpec(CommandLineInputSpec):
+    featDir = Directory(exists=True,
+                        desc='If a feat directory exists and temporal filtering '
+                        'has not been run yet, ICA_AROMA can use the files in '
+                        'this directory.',mandatory=False,xor=['infile','mask','affmat','warp','mc'])
+    infile = File(exists=True, 
+                  desc='volume to be denoised', 
+                  argstr='-i %s',mandatory=False,xor=['featDir'])
+    outdir = Directory(desc='path to output directory', 
+                  argstr='-o %s',mandatory=True)
+    mask = File(exists=True, 
+                desc='path/name volume mask', 
+                argstr='-m %s',mandatory=False,xor=['featDir'])
+    dim = traits.Int(desc='Dimensionality reduction when running '
+                     'MELODIC (defualt is automatic estimation)', 
+                     argstr='-dim %d')
+    TR = traits.Float(desc='TR in seconds. If this is not specified '
+                      'the TR will be extracted from the header of the fMRI nifti file.',
+                      argstr='%.2f')
+    melodic_dir = Directory(exists=True,
+                            desc='path to MELODIC directory if MELODIC has already been ran',
+                            argstr='-meldir %s')
+    affmat = File(exists=True,
+                  desc='path/name of the mat-file describing the '
+                  'affine registration (e.g. FSL FLIRT) of the '
+                  'functional data to structural space (.mat file)',
+                  argstr='-affmat %s',mandatory=False,xor=['featDir'])
+    warp = File(exists=True,
+                desc='File name of the warp-file describing '
+                'the non-linear registration (e.g. FSL FNIRT) '
+                'of the structural data to MNI152 space (.nii.gz)',
+                argstr='-warp %s',mandatory=False,xor=['featDir'])
+    mc = File(exists=True,
+              desc='motion parameters file',
+              argstr='-mc %s',mandatory=False,xor=['featDir'])
+    denoise_type = traits.Str(argstr='-den %s',
+                              desc='Type of denoising strategy: '
+                              '-none: only classification, no denoising '
+                              '-nonaggr (default): non-aggresssive denoising, i.e. partial component regression '
+                              '-aggr: aggressive denoising, i.e. full component regression '
+                              '-both: both aggressive and non-aggressive denoising (two outputs)',
+                              mandatory=True)
+
+class ICA_AROMAOutputSpec(TraitedSpec):
+    out_file = OutputMultiPath(File(exists=True), 
+                               desc='if generated: 1-(non aggressive denoised volume),'
+                               '2-(aggressive denoised volume)')
```

Comment:
This is a point where you can do the `OutputMultiPath` if you want, but I think having your original `aggr_denoised_file` and `nonaggr_denoised_file` (or similar) would be better, just because names are easier to deal with than indices.

For similar reasons, you'll also want anything that's in the `out_dir`. Looking through the source, I see:

* [`mask_file`](https://github.com/rhr-pruim/ICA-AROMA/blob/master/ICA_AROMA.py#L164-L185)
* [`melodic_dir`](https://github.com/rhr-pruim/ICA-AROMA/blob/master/ICA_AROMA_functions.py#L35)
* [`melodic_ic_thr`](https://github.com/rhr-pruim/ICA-AROMA/blob/master/ICA_AROMA_functions.py#L119-L122) (I don't know enough about this file to have a more descriptive name)
* [`melodic_ic_mni`](https://github.com/rhr-pruim/ICA-AROMA/blob/master/ICA_AROMA.py#L196) (same)
* [`feature_scores`](https://github.com/rhr-pruim/ICA-AROMA/blob/master/ICA_AROMA_functions.py#L482)
* [`classified_motion_components`](https://github.com/rhr-pruim/ICA-AROMA/blob/master/ICA_AROMA_functions.py#L485)
* [`classification_overview`](https://github.com/rhr-pruim/ICA-AROMA/blob/master/ICA_AROMA_functions.py#L491)

You could also do `out_dir`, if that seems reasonable (I see [MELODIC does this](https://github.com/nipy/nipype/blob/e7e2c002133c99fac47aafa2d180bb524546c220/nipype/interfaces/fsl/model.py#L1552-L1554)).